plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -622,7 +745,13 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
     @Override
     public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
```

Comment:
By adding row-level deletes, now some behaviors of `DELETE` becomes unclear to users. Some queries run as metadata delete, some as row-level delete. We might want to introduce the following configs:
1. if metadata delete should be used when possible
2. for metadata delete, should it use `DeleteFiles` API or global equality delete
3. for row-level delete, should it use position delete or equality delete

We know Trino architecture fits position delete better than equality delete, but please read document https://github.com/apache/iceberg/pull/3432 for the tradeoff between the 2. There might be use cases that people desire to use equality delete instead.

We will probably add these configs when the feature request comes.

Reply:
I think it's better to avoid new setting if possible. So questions:

1. If metadata delete is possible you'd always want that, right?
2. If so, why would anybody want the equality delete when DeleteFiles works?
3. Not clear here. That's seems to be the only one we want to configure. And probably on a query-by-query basis.

Reply:
1. yes, we should extend the feature of metadata delete to also consider Iceberg hidden partitions. As of today it only works for direct partition columns, and deleting based on partition transform predicate would result in row-level delete. That is the "unclear behavior to user" I meant.
2. because it does not touch data file, so will not cause commit conflict.
3. agree, I think we will need session config and ideally per-query config for this (if there is a need from community). But for now position delete definitely works the best with Trino row-level delete design.